### PR TITLE
[nfb] Add extractor

### PR DESCRIPTION
**Pull request by ofkz**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Added extractor for the National FIlm Board of Canada (https://www.nfb.ca/)

### CODE REVIEW

File: yt_dlp/extractor/nfb.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class NFBIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?nfb\.ca/(?:film)/(?P<id>[^/?#&]+)'
```
Comment (Ashish0804) [inline yt_dlp/extractor/nfb.py:8]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?nfb\.ca/film/(?P<id>[^/?#&]+)'
```

File: yt_dlp/extractor/nfb.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    _TESTS = [{
+        'url': 'https://www.nfb.ca/film/trafficopter/',
+        "info_dict": {
+            "id": "trafficopter",
+            'ext': 'mp4',
+            "title": "Trafficopter",
+            "description": "md5:060228455eb85cf88785c41656776bc0",
+            'thumbnail': r're:^https?://.*\.jpg$',
+            "uploader": "Barrie Howells",
+            "release_date": "1972",
```
Comment (Ashish0804) [inline yt_dlp/extractor/nfb.py:18]: Wrong field

File: yt_dlp/extractor/nfb.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        player = self._download_webpage(iframe, video_id)
+
+        source = self._html_search_regex(
+            r'source:\s*\'([^\']+)',
+            player, 'source', default=None, fatal=True)
+        thumbnail = self._html_search_regex(
+            r'poster:\s*\'([^\']+)',
+            player, 'thumbnail', default=None)
+
+        formats = self._extract_m3u8_formats(source, video_id, ext='mp4')
```
Comment (Ashish0804) [inline yt_dlp/extractor/nfb.py:54]: Use self._extract_m3u8_formats_and_subtitles

File: yt_dlp/extractor/nfb.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        return {
+            'id': video_id,
```
Comment (Ashish0804) [inline yt_dlp/extractor/nfb.py:57]: Inline title, description, thumbnail, uploader and sort formats before returning.

Comment (ofkz) [reply yt_dlp/extractor/nfb.py:58]: Make inline by moving those attributes into the dictionary?

Comment (Ashish0804) [reply yt_dlp/extractor/nfb.py:58]: Yes.

File: yt_dlp/extractor/nfb.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        'info_dict': {
+            'id': 'trafficopter',
+            'ext': 'mp4',
+            'title': 'Trafficopter',
+            'description': 'md5:060228455eb85cf88785c41656776bc0',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': 'Barrie Howells',
+            'release_date': '1972',
+        },
+    }
```
Comment (pukkandan) [inline yt_dlp/extractor/nfb.py:20]:
```suggestion
    _TESTS = [{
        'url': 'https://www.nfb.ca/film/trafficopter/',
        'info_dict': {
            'id': 'trafficopter',
            'ext': 'mp4',
            'title': 'Trafficopter',
            'description': 'md5:060228455eb85cf88785c41656776bc0',
            'thumbnail': r're:^https?://.*\.jpg$',
            'uploader': 'Barrie Howells',
            'release_date': '1972',
        },
    }]
```

File: yt_dlp/extractor/nfb.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage('https://www.nfb.ca/film/%s/' % video_id, video_id)
+
+        iframe = self._html_search_regex(
+            r'<[^>]+\bid=["\']player-iframe["\'][^>]*src=["\']([^"\']+)',
+            webpage, 'iframe', default=None, fatal=True)
+        if iframe.startswith("/"):
```
Comment (pukkandan) [inline yt_dlp/extractor/nfb.py:30]: single quotes everywhere

File: yt_dlp/extractor/nfb.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            iframe = 'https://www.nfb.ca' + iframe
```
Comment (pukkandan) [inline yt_dlp/extractor/nfb.py:31]:
```suggestion
            iframe = f'https://www.nfb.ca{iframe}'
```

File: yt_dlp/extractor/nfb.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+            'description': self._html_search_regex(
+                r'<[^>]+\bid=["\']tabSynopsis["\'][^>]*>\s*<p[^>]*>\s*([^<]+)',
+                webpage, 'description', default=None),
+            'thumbnail': self._html_search_regex(
+                r'poster:\s*\'([^\']+)',
+                player, 'thumbnail', default=None),
+            'uploader': self._html_search_regex(
+                r'<[^>]+\bitemprop=["\']name["\'][^>]*>([^<]+)',
+                webpage, 'uploader', default=None),
+            'release_date': self._html_search_regex(
```
Comment (pukkandan) [inline yt_dlp/extractor/nfb.py:56]: unified_strdate

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/nfb.py (hunk 8 above)
Comment (pukkandan) [reply yt_dlp/extractor/nfb.py:56]: actually, this seems to be just year? Use `release_year` then

**Pull request merged**